examples/openvino/aot/aot_openvino_compiler.py

```diff
@@ -0,0 +1,388 @@
+            raise ValueError(msg)
+        return getattr(torchvision_models, model_name)(pretrained=True)
+    elif suite == "huggingface":
+        return AutoModel.from_pretrained(model_name)
+    else:
+        msg = f"Unsupported model suite: {suite}"
+        raise ValueError(msg)
+
+
+def load_calibration_dataset(
+    dataset_path: str,
+    batch_size: int,
+    suite: str,
+    model: torch.nn.Module,
+    model_name: str,
+):
+    """
+    Loads a calibration dataset for model quantization.
+
+    :param dataset_path: Path to the dataset directory.
+    :param batch_size: Number of samples per batch.
+    :param suite: The model suite used for preprocessing transformations. Supported values are:
+        - "torchvision": Uses predefined transformations for torchvision models.
+        - "timm": Uses dataset transformations based on the model's pretrained configuration.
+    :param model: The model instance, required for timm transformation resolution.
+    :param model_name: The model name, required for torchvision transformations.
+    :return: A DataLoader instance for the calibration dataset.
+    :raises ValueError: If the suite is unsupported for validation.
+    """
+    val_dir = f"{dataset_path}/val"
+
+    if suite == "torchvision":
+        transform = torchvision_models.get_model_weights(
+            model_name
+        ).DEFAULT.transforms()
+    elif suite == "timm":
+        transform = create_transform(
+            **resolve_data_config(model.pretrained_cfg, model=model)
+        )
+    else:
+        msg = f"Validation is not supported yet for the suite {suite}"
+        raise ValueError(msg)
+
+    val_dataset = datasets.ImageFolder(val_dir, transform=transform)
+
+    calibration_dataset = torch.utils.data.DataLoader(
+        val_dataset,
+        batch_size=batch_size,
+        shuffle=False,
+        num_workers=0,
+        pin_memory=True,
+    )
+
+    return calibration_dataset
+
+
+def dump_inputs(calibration_dataset, dest_path):
+    """
+    Dumps the input data from a calibration dataset to raw files.
+
+    :param calibration_dataset: The dataset containing calibration inputs.
+    :param dest_path: The destination directory to save the raw input files.
+    :return: A tuple containing a list of input file paths and the corresponding target labels.
+    """
+    input_files, targets = [], []
+    for idx, data in enumerate(calibration_dataset):
+        feature, target = data
+        targets.extend(target)
+        file_name = f"input_{idx}_0.raw"
+        file_path = f"{dest_path}/{file_name}"
+        if not isinstance(feature, torch.Tensor):
+            feature = torch.tensor(feature)
+        feature.detach().numpy().tofile(file_path)
+        input_files.append(file_name)
+
+    return input_files, targets
+
+
+def quantize_model(
+    captured_model: torch.fx.GraphModule,
+    calibration_dataset: torch.utils.data.DataLoader,
+    use_nncf: bool,
```

Comment:
Why do we have this selection here with `use_nncf` and why it is not True by default?

Reply:
Taking into account that `quantize_model` will be moved to `quantizer.py`. I would suggest to remove `use_nncf` argument and keep only NNCF quantization flow in the function.